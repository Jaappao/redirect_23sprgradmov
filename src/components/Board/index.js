// @flow

import React from 'react'
import TimeRow from '../TimeRow'
import type { Period } from '../../types'
import config from '../../config'

import styled from 'styled-components'

const Wrapper = styled.div`
	padding: 10px;
	background: ${config.color.sub};
`
const Board = ({ periods }: { periods: Period[] }) => {
	return (
		<Wrapper>
			<p>開館 07:30</p>
			{periods.map(period => (
				<div>
					<TimeRow
						nextBreak={[2, 5].includes(period.info.period)}
						key={period.info.period}
						period={period}
					/>
				</div>
			))}
			<p>閉館 22:30</p>
		</Wrapper>
	)
}

export default Board
