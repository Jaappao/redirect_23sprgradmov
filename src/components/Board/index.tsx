import React from 'react'
import styled from 'styled-components'
import TimeRow from '../TimeRow'
import { Period } from '../../types'
import config from '../../config'

const Style = styled.div`
	padding: 10px;
	background: ${config.color.sub};

	display: flex;
	justify-content: center;
	> div {
		min-width: 300px;
		width: 100%;
		max-width: 600px;
		> p {
			font-weight: 800;
		}
	}
`

const Board = ({ periods }: { periods: Period[] }) => {
	return (
		<Style>
			<div>
				{periods.map((period, i) => (
					<TimeRow key={i} period={period} />
				))}
			</div>
		</Style>
	)
}

export default Board
