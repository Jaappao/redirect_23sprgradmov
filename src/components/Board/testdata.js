import moment from 'moment'

const periodsData = [
	{
		info: {
			period: '1',
			start: { h: 9, m: 20 },
			end: { h: 11, m: 0 },
		},
		status: { type: 'finish' },
		start: '2018-04-17T00:20:00.000Z',
		end: '2018-04-17T02:00:00.000Z',
	},
	{
		info: {
			period: '2',
			start: { h: 11, m: 10 },
			end: { h: 12, m: 50 },
		},
		status: { type: 'finish' },
		start: '2018-04-17T02:10:00.000Z',
		end: '2018-04-17T03:50:00.000Z',
	},
	{
		info: {
			period: '3',
			start: { h: 13, m: 40 },
			end: { h: 15, m: 20 },
		},
		status: { type: 'progress', progress: 75, rate: 0.75 },
		start: '2018-04-17T04:40:00.000Z',
		end: '2018-04-17T06:20:00.000Z',
	},
	{
		info: {
			period: '4',
			start: { h: 15, m: 30 },
			end: { h: 17, m: 10 },
		},
		status: { type: 'before', fromNowStr: '35分後' },
		start: '2018-04-17T06:30:00.000Z',
		end: '2018-04-17T08:10:00.000Z',
	},
	{
		info: {
			period: '5',
			start: { h: 17, m: 20 },
			end: { h: 19, m: 0 },
		},
		status: { type: 'before', fromNowStr: '2時間後' },
		start: '2018-04-17T08:20:00.000Z',
		end: '2018-04-17T10:00:00.000Z',
	},
	{
		info: {
			period: 'D5',
			start: { h: 18, m: 0 },
			end: { h: 19, m: 40 },
		},
		status: { type: 'before', fromNowStr: '3時間後' },
		start: '2018-04-17T09:00:00.000Z',
		end: '2018-04-17T10:40:00.000Z',
	},
	{
		info: {
			period: 'D6',
			start: { h: 19, m: 50 },
			end: { h: 21, m: 30 },
		},
		status: { type: 'before', fromNowStr: '5時間後' },
		start: '2018-04-17T10:50:00.000Z',
		end: '2018-04-17T12:30:00.000Z',
	},
]

const periods = periodsData.map(p => ({
	...p,
	start: moment(p.start),
	end: moment(p.end),
}))

export default periods
