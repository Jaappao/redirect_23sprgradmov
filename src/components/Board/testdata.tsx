import { Period } from '../../types'

const periodsData: Period[] = [
	{
		info: {
			period: '1',
			start: { h: 9, m: 20 },
			end: { h: 11, m: 0 },
		},
		status: { type: 'finish' },
	},
	{
		info: {
			period: '2',
			start: { h: 11, m: 10 },
			end: { h: 12, m: 50 },
		},
		status: { type: 'finish' },
	},
	{
		info: {
			period: '3',
			start: { h: 13, m: 40 },
			end: { h: 15, m: 20 },
		},
		status: { type: 'progress', progress: 75, rate: 0.75 },
	},
	{
		info: {
			period: '4',
			start: { h: 15, m: 30 },
			end: { h: 17, m: 10 },
		},
		status: { type: 'before' },
	},
	{
		info: {
			period: '5',
			start: { h: 17, m: 20 },
			end: { h: 19, m: 0 },
		},
		status: { type: 'before' },
	},
	{
		info: {
			period: 'D5',
			start: { h: 18, m: 0 },
			end: { h: 19, m: 40 },
		},
		status: { type: 'before' },
	},
	{
		info: {
			period: 'D6',
			start: { h: 19, m: 50 },
			end: { h: 21, m: 30 },
		},
		status: { type: 'before' },
	},
]

const periods = periodsData.map(p => ({
	...p,
}))

export default periods
