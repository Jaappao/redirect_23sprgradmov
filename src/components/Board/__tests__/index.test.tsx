import React from 'react'

import renderer from 'react-test-renderer'
import Board from '../'
import periods from './testdata'

import 'jest-styled-components'

test('renders correctly', () => {
	const tree = renderer.create(<Board periods={periods} />).toJSON()

	expect(tree).toMatchSnapshot()
})
