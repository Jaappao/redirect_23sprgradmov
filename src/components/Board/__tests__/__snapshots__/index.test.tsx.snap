// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=""
>
  <div>
    <p
      className=""
    >
       8:00
    </p>
    <div
      className=""
      data-study={true}
    >
      <div
        className="row"
      >
        <div
          className="label"
        >
          1
          .
        </div>
        <div
          className=""
        >
           9:20
           - 
          11:00
        </div>
        <div
          className=""
        />
        <div
          className=""
        >
          <div
            className=""
            color="gray"
          >
            <div>
              Fin
            </div>
          </div>
        </div>
      </div>
      <div
        className=""
      >
        <div
          className=""
        />
      </div>
    </div>
    <div
      className=""
      data-study={true}
    >
      <div
        className="row"
      >
        <div
          className="label"
        >
          2
          .
        </div>
        <div
          className=""
        >
          11:10
           - 
          12:50
        </div>
        <div
          className=""
        />
        <div
          className=""
        >
          <div
            className=""
            color="gray"
          >
            <div>
              Fin
            </div>
          </div>
        </div>
      </div>
      <div
        className=""
      >
        <div
          className=""
        />
      </div>
    </div>
    <div
      className=""
      data-study={false}
    >
      <div
        className="row"
      >
        <div
          className="label"
        >
          3
          .
        </div>
        <div
          className=""
        >
          13:40
           - 
          15:20
        </div>
        <div
          className=""
        >
          <span>
            (
            25
            min)
          </span>
        </div>
        <div
          className=""
        >
          <div
            className=""
            color="#ebe971"
          >
            <div>
              Now
            </div>
          </div>
        </div>
      </div>
      <div
        className=""
      >
        <div
          className=""
        />
      </div>
    </div>
    <div
      className=""
      data-study={true}
    >
      <div
        className="row"
      >
        <div
          className="label"
        >
          4
          .
        </div>
        <div
          className=""
        >
          15:30
           - 
          17:10
        </div>
        <div
          className=""
        />
        <div
          className=""
        >
          <div
            className=""
            color="black"
          >
            <div>
              
            </div>
          </div>
        </div>
      </div>
      <div
        className=""
      >
        <div
          className=""
        />
      </div>
    </div>
    <div
      className=""
      data-study={false}
    >
      <div
        className="row"
      >
        <div
          className="label"
        >
          5
          .
        </div>
        <div
          className=""
        >
          17:20
           - 
          19:00
        </div>
        <div
          className=""
        />
        <div
          className=""
        >
          <div
            className=""
            color="black"
          >
            <div>
              
            </div>
          </div>
        </div>
      </div>
      <div
        className=""
      >
        <div
          className=""
        />
      </div>
    </div>
    <div
      className=""
      data-study={true}
    >
      <div
        className="row"
      >
        <div
          className="label"
        >
          d5
          .
        </div>
        <div
          className=""
        >
          18:00
           - 
          19:40
        </div>
        <div
          className=""
        />
        <div
          className=""
        >
          <div
            className=""
            color="black"
          >
            <div>
              
            </div>
          </div>
        </div>
      </div>
      <div
        className=""
      >
        <div
          className=""
        />
      </div>
    </div>
    <div
      className=""
      data-study={false}
    >
      <div
        className="row"
      >
        <div
          className="label"
        >
          d6
          .
        </div>
        <div
          className=""
        >
          19:50
           - 
          21:30
        </div>
        <div
          className=""
        />
        <div
          className=""
        >
          <div
            className=""
            color="black"
          >
            <div>
              
            </div>
          </div>
        </div>
      </div>
      <div
        className=""
      >
        <div
          className=""
        />
      </div>
    </div>
    <p
      className=""
    >
      閉館 22:30
    </p>
  </div>
</div>
`;
